src/toast/ops/flag_sso.py

```diff
@@ -0,0 +1,272 @@
+
+    view = Unicode(
+        None, allow_none=True, help="Use this view of the data in all observations"
+    )
+
+    detector_pointing = Instance(
+        klass=Operator,
+        allow_none=True,
+        help="Operator that translates boresight Az/El pointing into detector frame",
+    )
+
+    det_flags = Unicode(
+        obs_names.det_flags,
+        allow_none=True,
+        help="Observation detdata key for flags to use",
+    )
+
+    det_flag_mask = Int(1, help="Bit mask to raise flags with")
+
+    sso_names = List(
+        default_value=["Sun", "Moon"],
+        trait=Unicode,
+        allow_none=True,
+        help="Names of the SSOs, must be recognized by pyEphem",
+    )
+
+    # FIXME: Once TOAST supports Lists of Quantities,
+    # FIXME: we can use astropy units here
+    sso_radii_deg = List(
+        default_value=["45.0", "5.0"],
+        trait=Unicode,  # Only type supported by TOAST for now
+        allow_none=True,
+        help="Radii (in degrees) around the sources to flag",
+    )
+
+    @traitlets.validate("detector_pointing")
+    def _check_detector_pointing(self, proposal):
+        detpointing = proposal["value"]
+        if detpointing is not None:
+            if not isinstance(detpointing, Operator):
+                raise traitlets.TraitError(
+                    "detector_pointing should be an Operator instance"
+                )
+            # Check that this operator has the traits we expect
+            for trt in [
+                "view",
+                "boresight",
+                "shared_flags",
+                "shared_flag_mask",
+                "quats",
+                "coord_in",
+                "coord_out",
+            ]:
+                if not detpointing.has_trait(trt):
+                    msg = f"detector_pointing operator should have a '{trt}' trait"
+                    raise traitlets.TraitError(msg)
+        return detpointing
+
+    @traitlets.validate("det_flag_mask")
+    def _check_det_flag_mask(self, proposal):
+        check = proposal["value"]
+        if check < 0:
+            raise traitlets.TraitError("Flag mask should be a positive integer")
+        return check
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def _exec(self, data, detectors=None, **kwargs):
+        env = Environment.get()
+        log = Logger.get()
+
+        if len(self.sso_names) != len(self.sso_radii_deg):
+            raise RuntimeError("Each SSO must have a radius")
+
+        self.ssos = []
+        for sso_name in self.sso_names:
+            self.ssos.append(getattr(ephem, sso_name)())
+        self.nsso = len(self.ssos)
+
+        for obs in data.obs:
+            dets = obs.select_local_detectors(detectors)
+            if len(dets) == 0:
+                # Nothing to do for this observation
+                continue
+            site = obs.telescope.site
+            observer = ephem.Observer()
```

Comment:
I know this work is porting existing code, but am looking forward to someday when we can convert to using astropy for coordinate transforms.  Note that a start on that work exists for the transform from Az/El to RA/DEC in `toast.coordinates`, but not yet the inverse transform.  And more testing needs to be done before switching completely from pyephem.  At some point pyephem will simply stop working with the latest python versions since it is no longer maintained.